app/controllers/ticket_purchases_controller.rb

```diff
@@ -11,8 +11,13 @@ def create
         redirect_to new_conference_payment_path,
                     notice: 'Please pay here to get tickets.'
       else
-        redirect_to conference_tickets_path(@conference.short_title),
-                    error: 'Please get at least one ticket to continue.'
+        if current_user.ticket_purchases.by_conference(@conference).paid.any?
+          redirect_to conference_conference_registration_path(@conference.short_title),
+                      notice: 'You have free tickets for the conference.'
```

Comment:
This notice will only be displayed if the user chooses ONLY "free tickets" out of all available tickets.
I am adding this message as they may be getting those "free tickets" for the second time and something like "You have successfully registered" can be confusing in that case